compiler/noirc_evaluator/src/ssa/ssa_gen/mod.rs

```diff
@@ -633,6 +637,25 @@ impl<'a> FunctionContext<'a> {
         Ok(result)
     }
 
+    /// If the condition is known, skip codegen for the then/else branch and only compile the
+    /// relevant branch.
+    fn try_codegen_constant_if(
+        &mut self,
+        condition: ValueId,
+        if_expr: &ast::If,
+    ) -> Option<Result<Values, RuntimeError>> {
+        let condition = self.builder.current_function.dfg.get_numeric_constant(condition)?;
```

Comment:
```suggestion
        let condition = !self.builder.current_function.dfg.is_constant_true(condition);
```

Reply:
You could use this method then and get rid of the import

Reply:
Probably cleaner to just do `let condition = self.builder.current_function.dfg.is_constant_true(condition);` and flip the if statement below.

Reply:
This doesn't differentiate between false constants and non-constants.

Reply:
Ah yes nevermind then